### DISCUSSION TITLE
[Feature] Sign in to Multiple Accounts on Mobile App (Account Switching)

### DISCUSSION THREAD

**Discussion by Tyree** [immich-app/immich#4718] [Feature Request]
### The feature

Similar to how Google Photos works, a way to sign in to multiple accounts and switch between them. I use Immich for myself and family, but also for my office. I use auto-upload for all my photos to my personal account, but I'd also like to be able to manually upload to my office account for photos I take for work.

As far as I can tell, currently, I have to sign out of one account and sign into the other. I'm not sure how settings like auto-upload transfer between accounts. I definitely do not want all my photos being auto-uploaded to my office account. I've avoided signing into my office account for fear the auto-upload settings would kick in and start uploading everything. :-)
So, like Google Photos, the settings for auto-upload should allow for choosing which account(s) you want to auto-upload to.

Thanks!

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by kameleon25** [comment]
I second this. Would be a great feature to have especially since the mobile web site doesn't tile the pictures very well (1 at a time rather than multiple per line).

**Comment by JonasMayerDev** [comment]
I would love this feature too! I think on mobile, the auto-upload is currently not independently saved for every user. For account switching, it would also be important to not re-upload every image and video again after logging in with an account with auto-upload enabled (seems to be the behavior right now on android). Maybe there is a way to check if a file is already uploaded? Thank you for the work you all put into this! :)

**Comment by kamelohr** [comment]
I just searched for this feature and found this thread. 

Also, it would be great to have not only the ability to add/switch accounts but also have multiple accounts from different servers (in case the other posts were referring to different accounts on the same host; it's not that clear to me)

Is there anything new related to this request?

**Reply by Tyree** [reply]
I was thinking different accounts in whatever form that takes, same server or different servers. In my case it would be for different servers, work and home.

I'm not sure if this has made it to the official roadmap or not.

**Comment by pelrun** [comment]
Agreed - I can handle the clunky account switching for the most part, but I'm forced to leave backup completely disabled so that it doesn't start uploading to the wrong account. If the setting was attached to the login then at least I could configure different folders for each account and not worry about cross contamination.

**Comment by bo0tzz** [comment]
dupe: #2919

**Comment by brainexcerpts** [comment]
That would be most welcomed indeed!!

**Comment by eric487** [comment]
Same problem here: my daughter installed Immich to save her photos on a separate account, but when she logged on to the family account, all her photos were automatically uploaded a second time to the family account! For the moment we've deactivated the automatic backup, but it would be much cleaner if the backup settings, and even others, were saved per account rather than globally to the application.

**Reply by marvil6** [reply]
Absolutely! I also had this issue.

**Comment by lubbertkramer** [comment]
Is there any update regarding the issue with the backup not connected to the account but to the device?

**Comment by ericloyd** [comment]
Just discovered Immich a few weeks ago, finally got my server up and running, and was dismayed to find that I can't do multiple accounts like in Dropbox, Nextcloud, Gmail, and more.  Yes, I can sign out and back in as a different user as others have mentioned, but the sync and backup folder selection is app-wide, not specific to the login.  That means, again, as others have said, that my personal folders get backed up to my work account and vice versa.  So I manually sync.  Which is cumbersome, but workable.

Ideally, I'd like to be able to be signed in to multiple accounts simultaneously and select between them (as well as backup, sync, settings, etc) by clicking on my account name, similar to the way Gmail does it or Nextcloud, or any of the other file-syncing apps do it.

Thanks!

**Comment by marvil6** [comment]
I would also very much appreciate this! I made a similar feature for web at #24857, but it was rejected because it was too niche, which is understandable.

**Comment by teksander** [comment]
I believe this is a much needed feature.  I have  three massively different datasets (personal photos + professional photos + grandfather digitally remastered photos) which I don't want to mix into the same facial recognition. I need to be able to smoothly alternate between users. Click User icon -> Select another user

Another possibility is alternating between servers. Click User icon -> Select server. In this case, I would run one Immich instance for each dataset I have.

**Reply by marvil6** [reply]
Absolutely agreed. I made a workaround for Web, but still nothing for Mobile - the only solution I've had so far was to build the dev version of the app from source and install it alongside the Play Store version, but that only allows for two accounts.

**Reply by teksander** [reply]
You gave me an idea and a quick google search yielded this: https://appcloner.app/

I could solve my issue by running three immich instances, and cloning the app three times on android.

**Reply by marvil6** [reply]
Awesome! Hope that works. I tried an app cloner too (not that one) but it only allowed cloning certain apps like WhatsApp and Telegram; it unfortunately did not list Immich. Anyways, let us know how it goes with App Cloner so we can give it a shot. Thanks!

**Reply by taboo180** [reply]
> You gave me an idea and a quick google search yielded this: https://appcloner.app/
> 
> I could solve my issue by running three immich instances, and cloning the app three times on android.

Did you have any success with this?

I was able to clone the app and even login. But once I started putting pics into the second account via the web gui, whenever I'd open the cloned app it would start to show that there's pics now in that account and then immediately crash.

If you've figured out the particular options to use in appcloner to get it to not crash I'd appreciate it!

Thanks.

**Reply by wassupluke** [reply]
I bet if you're cloning it after local(mobile) database creation that would cause the issue because then you've got multiple instances referencing the same database? Maybe for each new instance, Settings>Sync Status>Reset SQLite Database? Though I don't know if that simply resets the existing database used by all local(mobile) instances, or creates a new database. 🤷🏻‍♂️
